File: Assignment_15/program4.c
/////////////////////////////////////////////////////////////////
//
// Name :          program4.c
// Discription :   Display Pattern 
// Input :         Integer
// Output :        Charecter
// Author :        Shivam Rajendra Kale
// Date :          22/05/2025
//
/////////////////////////////////////////////////////////////////

/*
    Input : row = 6  col = 5
    
    Output : *  *  *  *  *
             *  @  @  @  *
             *  @  @  @  *
             *  @  @  @  *
             *  @  @  @  *
             *  *  *  *  *
*/

#include<stdio.h>

void Pattern(int iRow, int iCol)
{
    int i = 0, j = 0;

    for(i = 1; i <= iRow; i++)
    {
        for(j = 1; j <= iCol; j++)
        {
            if((j == 1)||(j == iCol)||(i == 1)||(i == iRow))
            {
                printf("*\t");
            }  
            else
            {
                printf("@\t");
            }  
        }
        printf("\n");
    }
}

int main()
{
    int iValue1 = 0, iValue2 = 0;

    printf("Enter number of rows :");
    scanf("%d",&iValue1);

     printf("Enter number of column :");
    scanf("%d",&iValue2);

    Pattern(iValue1, iValue2);   //function call

    return 0;
}

File: Assignment_50/program4.java
//////////////////////////////////////////////////////////////////////////
//
// Name :          program4.java
// Discription :   use to Display pattern
// Input :         Integer
// Output :        Charecters
// Author :        Shivam Rajendra Kale
// Date :          15/10/2025
//
//////////////////////////////////////////////////////////////////////////

import java.util.*;

class pattern
{
    public void Display(int iRow, int iCol)
    {
        int i = 0, j = 0;
        
        for(i = 0; i < iRow; i++)
        {
            for(j = iCol-1; j >= 0; j--)
            {
                if(i == j || i == 0 || j == 0 || j == iRow-1 || i == iCol-1)
                {
                    System.out.print("*\t");
                }
                else if(i < j)
                {
                    System.out.print("#\t");
                }
                else
                {
                    System.out.print("$\t");
                }
            }
            System.out.println();
        }
    }
}

class program4
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter number of rows :");
        int iRow = sobj.nextInt();

        System.out.println("Enter number of column :");
        int iCol = sobj.nextInt();

        pattern pobj = new pattern();

        pobj.Display(iRow, iCol);

        sobj.close();
    }
}